apps/passport-client/src/dispatch.ts

```diff
@@ -756,6 +770,12 @@ async function doSync(
     state.pcds,
     state.subscriptions
   );
+  const validationErrors = validateUpload(state.self, state.pcds);
```

Comment:
Probably don't bother validating if the hash is unchanged below.
There's also a validation inside of uploadStorage().  The two could be combined if the code goes inside of uploadSerializedStorage().

Reply:
fixed